src/util/util_matrix.cpp

```diff
@@ -205,6 +205,10 @@ namespace dxvk {
     Vector4 dot0    = { m[0] * row0 };
     float dot1      = (dot0.x + dot0.y) + (dot0.z + dot0.w);
 
+    if (unlikely(std::abs(dot1) <= 0.000001f)) {
```

Comment:
which part of the API relies on this and do we have a way to verify native behaviour?

Reply:
It's specifically about [calculating the normal matrix from the world matrix](https://github.com/doitsujin/dxvk/blob/master/src/d3d9/d3d9_device.cpp#L7334).

Wine tests what D3D9 does with a singular matrix [here](https://github.com/wine-mirror/wine/blob/master/dlls/d3d9/tests/visual.c#L604).

Would you prefer it if I check the determinant separately even if that means calculating it twice?